pkg/server/plugin/datastore/sql/sql_test.go

```diff
@@ -2519,131 +2519,100 @@ func (s *PluginSuite) TestDeleteBundleDissociateRegistrationEntries() {
 }
 
 func (s *PluginSuite) TestCreateJoinToken() {
-	now := time.Now().Unix()
-	req := &datastore.CreateJoinTokenRequest{
-		JoinToken: &datastore.JoinToken{
-			Token:  "foobar",
-			Expiry: now,
-		},
+	req := &datastore.JoinToken{
+		Token:  "foobar",
+		Expiry: time.Now().Truncate(time.Second),
 	}
-	_, err := s.ds.CreateJoinToken(ctx, req)
+	err := s.ds.CreateJoinToken(ctx, req)
 	s.Require().NoError(err)
 
 	// Make sure we can't re-register
-	_, err = s.ds.CreateJoinToken(ctx, req)
+	err = s.ds.CreateJoinToken(ctx, req)
 	s.NotNil(err)
 }
 
 func (s *PluginSuite) TestCreateAndFetchJoinToken() {
-	now := time.Now().Unix()
+	now := time.Now().Truncate(time.Second)
 	joinToken := &datastore.JoinToken{
 		Token:  "foobar",
 		Expiry: now,
 	}
 
-	_, err := s.ds.CreateJoinToken(ctx, &datastore.CreateJoinTokenRequest{
-		JoinToken: joinToken,
-	})
+	err := s.ds.CreateJoinToken(ctx, joinToken)
 	s.Require().NoError(err)
 
-	res, err := s.ds.FetchJoinToken(ctx, &datastore.FetchJoinTokenRequest{
-		Token: joinToken.Token,
-	})
+	res, err := s.ds.FetchJoinToken(ctx, joinToken.Token)
 	s.Require().NoError(err)
-	s.Equal("foobar", res.JoinToken.Token)
-	s.Equal(now, res.JoinToken.Expiry)
+	s.Equal("foobar", res.Token)
+	s.Equal(now, res.Expiry)
 }
 
 func (s *PluginSuite) TestDeleteJoinToken() {
-	now := time.Now().Unix()
+	now := time.Now().Truncate(time.Second)
 	joinToken1 := &datastore.JoinToken{
 		Token:  "foobar",
 		Expiry: now,
 	}
 
-	_, err := s.ds.CreateJoinToken(ctx, &datastore.CreateJoinTokenRequest{
-		JoinToken: joinToken1,
-	})
+	err := s.ds.CreateJoinToken(ctx, joinToken1)
 	s.Require().NoError(err)
 
 	joinToken2 := &datastore.JoinToken{
 		Token:  "batbaz",
 		Expiry: now,
 	}
 
-	_, err = s.ds.CreateJoinToken(ctx, &datastore.CreateJoinTokenRequest{
-		JoinToken: joinToken2,
-	})
+	err = s.ds.CreateJoinToken(ctx, joinToken2)
 	s.Require().NoError(err)
 
-	_, err = s.ds.DeleteJoinToken(ctx, &datastore.DeleteJoinTokenRequest{
-		Token: joinToken1.Token,
-	})
+	err = s.ds.DeleteJoinToken(ctx, joinToken1.Token)
 	s.Require().NoError(err)
 
 	// Should not be able to fetch after delete
-	resp, err := s.ds.FetchJoinToken(ctx, &datastore.FetchJoinTokenRequest{
-		Token: joinToken1.Token,
-	})
+	resp, err := s.ds.FetchJoinToken(ctx, joinToken1.Token)
 	s.Require().NoError(err)
-	s.Nil(resp.JoinToken)
+	s.Nil(resp)
 
 	// Second token should still be present
-	resp, err = s.ds.FetchJoinToken(ctx, &datastore.FetchJoinTokenRequest{
-		Token: joinToken2.Token,
-	})
+	resp, err = s.ds.FetchJoinToken(ctx, joinToken2.Token)
 	s.Require().NoError(err)
-	s.Equal(joinToken2, resp.JoinToken)
+	s.Equal(joinToken2, resp)
 }
 
 func (s *PluginSuite) TestPruneJoinTokens() {
```

Comment:
Hmm, the usage of `time.Now()` throughout this method feels like a smell to me. The test used to at least base all its timestamp calculations based on a stable `now` established at the beginning of the test. Is there any reason why that behavior was changed?

Reply:
Usually using a mock clock is preferred for any tests that need to work with time, but I see this was an existing issue as well...

Reply:
good catch; that was a leftover from iterations of changing from int64 to time.Time